File: e2_CountTheHiddenSequences/typescript/Main.ts
import { Solution } from './Solution';

function main() {
    const solution = new Solution();
    const differences = [1, -3, 4];
    const lower = 1;
    const upper = 6;
    const result = solution.numberOfArrays(differences, lower, upper);

    console.log("Input differences: [" + differences.join(", ") + "]");
    console.log("Input lower: " + lower);
    console.log("Input upper: " + upper);
    console.log("Output: " + result); // Expected output for this example is 2
}

main();

File: e2_CountTheHiddenSequences/java/Main.java
package e2_CountTheHiddenSequences.java;

public class Main {
    public static void main(String[] args) {
        Solution_TimeExcepted solution = new Solution_TimeExcepted();
        int[] differences = {1, -3, 4};
        int lower = 1;
        int upper = 6;
        int result = solution.numberOfArrays(differences, lower, upper);
        System.out.println("Input differences: " + java.util.Arrays.toString(differences));
        System.out.println("Input lower: " + lower);
        System.out.println("Input upper: " + upper);
        System.out.println("Output: " + result);
    }
}
